modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/service/impl/MotechSchedulerServiceImpl.java

```diff
@@ -786,7 +797,7 @@ private void validateJob(JobKey key) throws SchedulerException {
 
         JobDataMap map = detail.getJobDataMap();
 
-        if (map != null && !map.getBooleanValue(UI_DEFINED)) {
+        if (map != null && map.get(EVENT_METADATA) != null && !(Boolean) ((Map<String, Object>) map.get(EVENT_METADATA)).get(UI_DEFINED)) {
```

Comment:
ok understand, please move this condition into method, because it is hard to understand at the first look